src/SpendPermissionManager.sol

```diff
@@ -194,13 +206,17 @@ contract SpendPermissionManager is EIP712 {
 
     /// @notice Approve a spend permission via a signature from the account.
     ///
+    /// @dev Compatible with ERC-6492 signatures (https://eips.ethereum.org/EIPS/eip-6492)
+    /// Validates a signature over spend permission data and optionally deploys account.
+    ///
     /// @param spendPermission Details of the spend permission.
     /// @param signature Signed approval from the user.
     function approveWithSignature(SpendPermission calldata spendPermission, bytes calldata signature) public {
-        // validate signature over spend permission data
+        // validate signature over spend permission data, deploying or preparing account if necessary
         if (
-            IERC1271(spendPermission.account).isValidSignature(getHash(spendPermission), signature)
-                != IERC1271.isValidSignature.selector
+            !erc6492Deployer.isValidERC6492SignatureNowAllowSideEffects(
```

Comment:
Given we're defining this method, I'd prefer something a bit more streamlined. Not happy with the new names coming to mind though, so may be fine with this for now, can sleep on it.